frontend/packages/git-service/src/utils/import-strategy-detector.ts

```diff
@@ -61,10 +70,15 @@ export const detectImportStrategies = async (
     : RepoStatus.GitTypeNotDetected;
   let detectedFiles: string[] = [];
   let detectedCustomData: string[];
+  let pacFiles: string[] = [];
+  let addPacRepositoryStrategy: boolean;
 
   if (repositoryStatus === RepoStatus.Reachable) {
     try {
-      const { files } = await gitService.getRepoFileList();
+      const { files } = await gitService.getRepoFileList({ includeFolder: true });
+      pacFiles = await detectPacFiles(gitService);
+      addPacRepositoryStrategy = await isPacRepository(isRepositoryEnabled, gitService, pacFiles);
```

Comment:
The additional `includeFolder` here is added so that you also get `.tekton`  from the getRepoFileList call, or?

But you didn't use this at the moment, which is why `detectPacFiles` tries to load files from `.tekton` also for repos without such a folder. (Saw 404 Not found in the network inspector.) I think you can improve this.

Reply:
The `includeFolder` has been incorporated to solve the issue with Gitlab APIs. Actually, the current implementation of `getRepoFileList` of Gitlab, only fetches filenames, not folders. Therefore, `.tekton` is not been fetched, which makes all Gitlab repos negative for PAC strategy in spite of having `.tekton` folders.